File: app/src/main/java/com/pkfcooperparry/template/data/local/db/DbHelper.java
package com.pkfcooperparry.template.data.local.db;

import com.pkfcooperparry.template.data.model.db.User;
import com.pkfcooperparry.template.data.model.db.visit.Form;
import com.pkfcooperparry.template.data.model.db.visit.Page;
import com.pkfcooperparry.template.data.model.db.visit.Tab;

import java.util.List;

import io.reactivex.Observable;

/**
 * Interface that will include all the DB methods
 */

public interface DbHelper {
    Observable<List<User>> getAllUsers();

    User getUser();

    void addUser(User user);

    void updateUser(User user, String firstName, String surname, int sex, int age,
                    String hometown, int positionId, String phone);

    void deleteAll();

    Tab getTab(long tabId);

    Form getFormWithId(int formId);

    Page getPageWithId(int pageId);

    List<Form> getForms();

    void addForms(List<Form> forms);

}

File: app/src/main/java/com/televantou/template/data/local/db/DbHelper.kt
package com.televantou.template.data.local.db

import com.televantou.template.data.model.db.User
import com.televantou.template.data.model.db.visit.Form
import com.televantou.template.data.model.db.visit.Page
import com.televantou.template.data.model.db.visit.Tab

import io.reactivex.Observable

/**
 * Interface that will include all the DB methods
 */

interface DbHelper {
    val allUsers: Observable<List<User>>

    val user: User?

    val forms: List<Form>

    fun addUser(user: User)

    fun updateUser(user: User, firstName: String, surname: String, sex: Int, age: Int,
                   hometown: String, positionId: Int, phone: String)

    fun deleteAll()

    fun getTab(tabId: Long): Tab?

    fun getFormWithId(formId: Int): Form?

    fun getPageWithId(pageId: Int): Page?

    fun addForms(forms: List<Form>)

}
